From 3c3c911526865c4971d1507c6ef52874bd4a562e Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Sun, 4 Oct 2015 11:50:13 +0000
Subject: [PATCH 0689/1077] MANTIS 0037349 change: integration of ingenico
 payment device for KIS kiosk

mStateMachine.hpp
V1.0.0.29

allow to set the log level by the transition


git-svn-id: svn://localhost/SelfServiceCommon/trunk@909 90b65887-3827-0410-9a23-83215b262276
---
 .../Interfaces/generic/fsm/mStateMachine.hpp   | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Interfaces/generic/fsm/mStateMachine.hpp b/SelfServiceCommon/Interfaces/generic/fsm/mStateMachine.hpp
index 6ecea5e0..4cfd06d1 100644
--- a/SelfServiceCommon/Interfaces/generic/fsm/mStateMachine.hpp
+++ b/SelfServiceCommon/Interfaces/generic/fsm/mStateMachine.hpp
@@ -233,6 +233,10 @@ namespace StateMachine
          return 0;
       }
 
+      virtual int getLoglevel() const
+      {
+         return 5;
+      }
    };
 
 
@@ -649,6 +653,18 @@ namespace StateMachine
          return M_TargetPlace;
       }
 
+      int getLoglevel() const
+      {
+         if(M_TransitionHandler)
+         {
+            return M_TransitionHandler->getLoglevel();
+         }
+         else
+         {
+            return 5;
+         }
+      }
+
    private:
 
       /**
@@ -1548,7 +1564,7 @@ bool Materna::StateMachine::place<EVT>::inputTransitions
                LogStream << " -> " << pTarget->name();
                Wlk.logging
                   (
-                     Wlk.Log_BasicRuntime,
+                     walker<EVT>::logLevel(TrI->getLoglevel()),
                      LogStream.str().c_str()
                   );
             }
-- 
2.41.0.windows.1

